docs/plugins.md

```diff
@@ -1 +1,124 @@
+
+For Kn we have adopted a similar model except our prefix for naming plugins is `kn-`. So a Kn plugin that is similar to the one discussed above would be named `kn-hello` and be invoked with `kn hello`. Naming conflicts also occur like for `kubectl`. 
+
+However, we have one main difference with `kubectl` plugins and that is the location of Kn plugins can be centralized to a user’s provided location. The default is `~/.kn/plugins` and can be specified in the Kn config file or via the command line using `—lookup-plugins—in-path`. 
+
+You can also override the location of plugins from the default of `~/.kn/plugins` to a different directory by passing the directory with `--plugins-dir` or setting that value in your Kn config. Read about [Kn’s config](docs/config.md) file for details on where it is located and the values you can change.
+
+Before showing some example Kn plugins in different languages, let’s first explore some use cases of possible Kn plugins.
+
+### Use Cases
+
+There are many use cases for Kn Plugins. We give a brief summary of five classes of use cases and some simple examples below to help you get started:
+
+1. *Workflow* to allow users of Knative to expand their current workflows in a natural fashion. For instance DevOps users that end up creating lots of custom scripts for recurring work items, e.g., setting up clusters with Knative on their private cloud, could easily create a `kn-setup` plugin that would blend into the current Kn CLI for ease of use.
+
+2. *Integration* to allow integration of Knative with other systems. For instance a set of plugins to facilitate an integrated and seamless usage of Knative with the [Tekton CI/CD](https://github.com/tektoncd/pipeline).
+
+3. *Extension* to add new features to Knative. For instance creating a `kn-logs` plugin that will use `kubectl` to discover and stream the logs from the pods of deployed and running Knative services.
+
+4. *Tooling* to provide utilities and tools for Knative. For instance to return and display the healthiness of a running Knative cluster using heuristics and the Kubernetes APIs.
+
+5. *Experimentation* to explore the boundaries of what _serverless_ applications and architectures are able to achieve when using Knative as their base platform. For example, creating a plugin to spin up Knative services across clusters in different geographic regions and load balancing taking into account request headers and geo locations.
+
+Naturally since plugins allow a wide range of extensions to Kn and Knative, from a user’s perspective, the use cases listed above constitute only a cross section of what is possible. As Knative and the Kn CLI gain users, we fully anticipate innovative and unexpected plugins to appear in the community.
+
+## Examples
```

Comment:
Could we put the examples to the end ? Or leave one example here and then add the other examples towards the end ? I think its more important to have the other plugin command `kn plugin list` more prominent before the full examples.

Reply:
Also I think it would be totally fine to have just a list of multiple language examples and just reference them from within this document.